api/src/environment.js

```diff
@@ -1,9 +1,24 @@
 const path = require('path');
 const logger = require('./logger');
 
-const { UNIT_TEST_ENV, COUCH_URL, BUILDS_URL } = process.env;
+const { 
+  UNIT_TEST_ENV,
+  COUCH_URL,
+  BUILDS_URL, 
+  // If set to true, PROXY_CHANGE_ORIGIN_ALL will override the values of PROXY_CHANGE_ORIGIN, 
+  // PROXY_CHANGE_ORIGIN_AUTH and PROXY_CHANGE_ORIGIN_CHANGES and set them to true.
+  PROXY_CHANGE_ORIGIN_ALL = false, 
+  PROXY_CHANGE_ORIGIN = false,
+  PROXY_CHANGE_ORIGIN_AUTH = false,
+  PROXY_CHANGE_ORIGIN_CHANGES = false
```

Comment:
Can you think of any time when you would want one proxy to have change origin but not the others? Because they're all going to proxy to the same couch instance the sni will either be required by all or by none. If you set one to true and the others to false you would end up with a really hard to debug setup. This complexity makes life much harder for implementers and unless there's a really good reason to include it we should aim to make their job as easy as possible.

Reply:
Well, the existence of the multiple clients means that conceptually they could be pointed at differently configured entities, but if you don't know of any use cases, I'm OK with collapsing them. 

My general bent is to expose configurability/apis where possible and to try not predict usage too much.

Will change.